pkg/controller/machine_safety.go

```diff
@@ -141,8 +141,9 @@ func (c *controller) checkAndFreezeORUnfreezeMachineSets(wg *sync.WaitGroup) {
 func (c *controller) checkVMObjects(wg *sync.WaitGroup) {
 	var wgForCheckVM sync.WaitGroup
 
-	wgForCheckVM.Add(1)
+	wgForCheckVM.Add(2)
 	go c.checkAWSMachineClass(&wgForCheckVM)
+	go c.checkOSMachineClass(&wgForCheckVM)
```

Comment:
Would be nice to have this as part of the doku on how to add support for new provider.

Reply:
Done.